doc/essentials/EnergyPlusEssentials.tex

```diff
@@ -0,0 +1,2260 @@
+is often the simulation engine for graphical user interfaces. To see
+a comprehensive list, see the \href{https://www.buildingenergysoftwaretools.com/}{BEST (Building Energy Software Tools) Directory}
+that is operated by \href{https://www.ibpsa.us/}{IBPSA-USA}. At the
+time of writing, many people use EnergyPlus inside an existing graphical
+user interface such as \href{https://www.openstudio.net/}{OpenStudio},
+\href{https://www.designbuilder.com/}{DesignBuilder}, \href{https://d-alchemy.com/products/simergy}{Simergy},
+\href{https://www.ladybug.tools/}{Ladybug}, \href{https://www.autodesk.com/products/insight/overview}{Insight},
+\href{https://www.bentley.com/en/products/brands/aecosim}{AECOsim},
+and \href{https://bigladdersoftware.com/projects/euclid/}{Euclid}.
+Other interfaces are also available. 
+
+\subsection{Approaches to Implement Measures }
+
+In the terminology used within the building energy modeling industry
+``measures,'' sometimes called energy conservation measures (ECMs)
+or energy efficiency measures (EEMs), are when alternative configurations
+of a building are considered and simulated and compared with the original
+building model. Measures include added wall or roof insulation, lower
+internal lighting power, higher efficiency heating and cooling equipment,
+and many others. When using EnergyPlus within a graphical user interface,
+that interface often provides a way to implement various measures.
+When not working within a graphical user interface, users have several
+options for implementing measures.
+
+For a specific building input file, a copy of the input file can be
+made and then modified to reflect the measure. This is an easy approach
+initially, but since the original building model might change, it
+means making the change in the file that reflects the measure. When
+many measures are being considered, this duplicative editing can be
+inefficient and prone to errors. In addition, the measure cannot be
+easily applied to a different building energy model without again
+editing another set of files duplicating the original effort. Due
+to this, it is very common for some type of scripting to be used so
+that measures can be applied to the original building model and other
+models of other buildings reliably and quickly. Scripting in this
+approach means to apply some type of programming language to the task
+of modifying files.
+
+EnergyPlus includes, with the installation package, two methods of
+implementing measures EP-Macro and the ParametricPreprocessor. EP-Macro
+is similar to the C language pre-processor and allows for portions
+of the file to be included or excluded, portions of other files to
+be included when desired, and for specific entries that are normally
+fixed values to change programmatically. EP-Macro is documented in
+the AuxiliaryPrograms documentation. A file with lines starting ``\#\#''
+or with the ``imf'' file extension are likely to be a file using
+EP-Macro files. 
```

Comment:
Would you consider ExpandObjects another way to implement measures?

Reply:
I don't usually, could you explain further?